### PR TITLE
Add contributing guidelines

### PR DESCRIPTION
**Pull request by rsanchez15**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+Contributors must sign-off each commit by adding a `Signed-off-by: ...`
+line to commit messages to certify that they have the right to submit
+the code they are contributing to the project according to the
+[Developer Certificate of Origin (DCO)](https://developercertificate.org/).
+
+## Code Coverage
+
+All contributions to the project must increase line code coverage.
+Because of this, contributors are asked to locally run a coverage assessment that ensures that code coverage has increased when compared to the latest execution of the [nightly coverage CI job](http://jenkins.eprosima.com:8080/job/nightly_fastdds_coverage_linux/).
```
Comment (jparisu) [inline CONTRIBUTING.md:36]: URL is not correct, our coverage is somewhere else

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+## Developer Certificate of Origin
+
+Contributors must sign-off each commit by adding a `Signed-off-by: ...`
+line to commit messages to certify that they have the right to submit
+the code they are contributing to the project according to the
+[Developer Certificate of Origin (DCO)](https://developercertificate.org/).
+
+## Code Coverage
+
+All contributions to the project must increase line code coverage.
```
Comment (jparisu) [inline CONTRIBUTING.md:35]: are we sure we want this?

**Review by jparisu — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (rsanchez15) [reply CONTRIBUTING.md:35]: Rephrased

**Review by jparisu — APPROVED** [review]
LGTM

**Pull request merged**